apps/api/src/modules/providers/providers.service.ts

```diff
@@ -64,8 +67,11 @@ export class ProvidersService extends CoreService<Provider> {
     });
 
     if (configEntity) {
-      this.logger.debug('config entry fetched successfully');
-      return configEntity.configuration as unknown as Record<string, unknown>;
+      this.logger.debug('Config entry fetched successfully');
+
+      // Decrypt the configuration before returning
+      const decryptedConfig = decrypt(configEntity.configuration);
```

Comment:
I am assuming all configurations are fetched using getConfigById? Make sure that is the case.

Reply:
yes